src/c_api/c_api.cc

```diff
@@ -497,12 +499,139 @@ XGB_DLL int XGDMatrixCreateFromMat_omp(const bst_float* data,  // NOLINT
       }
     }
   }
+  // restore omp state
+  omp_set_num_threads(nthread_orig);
 
   mat.info.num_nonzero_ = mat.page_.data.size();
   *out  = new std::shared_ptr<DMatrix>(DMatrix::Create(std::move(source)));
   API_END();
 }
 
+int DTGetType(const wchar_t* type_string) {
+  if (wcscmp(type_string, L"float32") == 0) {
```

Comment:
You should include header `<cwchar>` to use `wcscmp`. Alternatively, use `strcmp` from `<cstring>`. (To choose this alternative, make sure to obtain UTF-8 encoded strings from the Python end.)